src/java/org/apache/cassandra/service/accord/interop/AccordInteropApply.java

```diff
@@ -248,20 +245,14 @@ public void onChange(SafeCommandStore safeStore, SafeCommand safeCommand)
             case PreCommitted:
             case Committed:
             case PreApplied:
-                return;
+                return true;
 
             case Applied:
             case Invalidated:
             case Truncated:
         }
 
-        if (safeCommand.removeListener(this))
-            ack();
-    }
-
-    @Override
-    public PreLoadContext listenerPreLoadContext(TxnId caller)
-    {
-        return PreLoadContext.contextFor(txnId);
+        ack();
```

Comment:
We return `false` here, so the listener should be deregistered immediately and not invoked again. This is different to `ReadData`, which stays registered to be notified of being superseded (e.g. if the result is `Applied` before we finish, so that our read is now potentially invalid)